### ISSUE TITLE
Pylance does not report expected error "Argument missing"

### ISSUE THREAD

**Issue by PASUNX** [microsoft/pylance-release#1104]
<h3> Environment data </h3>

-   Language Server version: 2021.3.3
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: strict

## Code Snippet / Additional information

```python
class A:

    def __init__(self, x: str, *, y: str) -> None: ...

class B:

    def __init__(self, a: A) -> None: ...

def C(b: B) -> None: ...

B(A(y = "Y")) # error: Argument missing for parameter "x" Pylance (reportGeneralTypeIssues)

C(B(A(y = "Y"))) # no error
```
![image](https://user-images.githubusercontent.com/76944694/112967072-0a9a0800-9175-11eb-8db9-d848bca1712f.png)

**Comment by erictraut** [comment]
This will be fixed in the next release of pylance. Thanks for the bug report.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.3.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202134-31-march-2021
